pkg/executables/govc.go

```diff
@@ -130,6 +131,42 @@ func (g *Govc) LibraryElementExists(ctx context.Context, library string) (bool,
 	return response.Len() > 0, nil
 }
 
+func (g *Govc) deleteLibraryElement(ctx context.Context, element string) error {
+	_, err := g.exec(ctx, "library.rm", element)
+	if err != nil {
+		return fmt.Errorf("govc failed deleting library item: %v", err)
+	}
+
+	return nil
+}
+
+func (g *Govc) DeleteOVAIfInvalid(ctx context.Context, element string) (bool, error) {
```

Comment:
Imo this method does too many things. I think we should use the govc executable mostly as a wrapper around the govc binary operations

Could we just have a `GetLibraryElementInfo` method and make the template library decide if the element needs to be deleted? (calling delete library element)

In that case you could even skip the `LibraryElementExists` step, since `GetLibraryElementInfo` could return nil if it doesn't exist

You could just request the library.info output in json format and parse it into a struct (I think we do something similar for other vsphere types)